orttraining/orttraining/core/graph/zero_optimizer_graph_builder.cc

```diff
@@ -19,6 +21,96 @@ static bool IsNcclAvailable() {
 #endif
 }
 
+static int NumberOfWorkers(std::vector<int64_t>& size_arr, int64_t max_len, std::vector<int64_t>& partitions) {
```

Comment:
can size_arr be a const vector&? similar comment for other input vector parameters.